src/thresholds.service.js

```diff
@@ -30,30 +30,49 @@ class ThresholdsService {
   // For zones the thresholds are based on the state store required allocation for
   // the week, that information is passed as an optional param (`requiredStateStoresAllocation`).
   // That param is only used for zones.
-  calculateThresholds (location, stockCount, products, requiredStateStoresAllocation = {}) {
-    if (!location || !location.allocations || !location.plans || !location.level) {
+  calculateThresholds (location, stockCount, products, requiredStateStoresAllocation = {}, options = {}) {
+    if (!location || !location.allocations || !location.allocations.length ||
+      !location.plans || !location.plans.length || !location.level) {
       return
     }
 
-    if (!stockCount || !stockCount.allocations || !stockCount.allocations.version ||
-        !stockCount.plans || !stockCount.plans.version) {
+    if (!stockCount) {
+      return
+    }
+
+    if (options.version !== 'last' &&
+        !(stockCount.allocations && typeof stockCount.allocations.version !== undefined &&
+          stockCount.plans && typeof stockCount.plans.version !== undefined)) {
       return
     }
 
     if (!products || !products.length) {
       return
     }
 
-    const allocation = find(location.allocations, isVersion.bind(null, stockCount.allocations.version))
+    let allocation
+    if (options.version === 'last') {
+      allocation = location.allocations[location.allocations.length - 1]
+    } else {
+      allocation = find(location.allocations, isVersion.bind(null, stockCount.allocations.version))
```

Comment:
Agree, there is a separate ticket for this https://github.com/fielded/angular-nav-thresholds/issues/16